extensions/amp/bundle/src/main/java/com/adobe/cq/wcm/core/extensions/amp/internal/AmpModeForwardFilter.java

```diff
@@ -64,61 +66,24 @@ public void doFilter(final ServletRequest request, final ServletResponse respons
             throws IOException, ServletException {
 
         SlingHttpServletRequest slingRequest = (SlingHttpServletRequest) request;
+        AMP_MODE ampMode = getAmpMode(slingRequest);
 
-        // Read the full selector string from the request.
-        String selectors = slingRequest.getRequestPathInfo().getSelectorString();
-        if (selectors == null) {
-            selectors = "";
-        }
+        List<String> selectors = Arrays.asList(slingRequest.getRequestPathInfo().getSelectors());
```

Comment:
I would rather we create a modifiable list here: `List<String> selectors = new LinkedList(Arrays.asList(slingRequest.getRequestPathInfo().getSelectors()));`

Reply:
`Arrays.asList` will return a modifiable ArrayList. I don't see a benefit of a LinkedList here.

Reply:
https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#asList(T...)
> Returns a fixed-size list backed by the specified array
That's read-only.